yoda/yoda.h

```diff
@@ -199,26 +120,25 @@ class API {
         state_maintainer_(stream_),
         listener_scope_(stream_.Subscribe(state_maintainer_)) {}
 
-  typename sherlock::StreamInstance<T_ENTRY>& UnsafeStream() { return stream_; }
+  T_STREAM_TYPE& UnsafeStream() { return stream_; }
 
   template <typename F>
-  typename StreamInstanceImpl<T_ENTRY>::template ListenerScope<F> Subscribe(F& listener) {
+  T_STREAM_LISTENER_TYPE<F> Subscribe(F& listener) {
     return std::move(stream_.Subscribe(listener));
   }
 
-  std::future<T_ENTRY> AsyncGet(const T_KEY&& key) {
+  std::future<T_ENTRY> AsyncGet(const T_KEY& key) {
     std::promise<T_ENTRY> pr;
     std::future<T_ENTRY> future = pr.get_future();
     state_maintainer_.mq_.EmplaceMessage(new MQMessageGet(key, std::move(pr)));
     return future;
   }
 
-  void AsyncGet(const T_KEY&& key, T_ENTRY_CALLBACK on_success, T_KEY_CALLBACK on_failure = [](const T_KEY&) {
-  }) {
+  void AsyncGet(const T_KEY& key, T_ENTRY_CALLBACK on_success, T_KEY_CALLBACK on_failure) {
```

Comment:
Yes, I gave it another thought and realized it's a bad idea.

If, say, there's an HTTP endpoint, and the user simply forgets the last parameter, those HTTP requests will get hanging forever. It's a hard to catch bug. Not good, better eliminate all the room for it altogether.